modules/aioseop_sitemap.php

```diff
@@ -1027,9 +1027,12 @@ function get_problem_files( $files, &$msg ) {
 						}
 					} else {
 						if ( $compressed ) {
-							$fn = 'compress.zlib://' . $fn;
+							$file_resource = gzopen( $fn, 'rb' );
+							$file = gzread( $file_resource, 4096 );
+							gzclose( $file_resource );
+						} else {
+							$file = file_get_contents( $fn, false, null, 0, 4096 );
```

Comment:
@EkoJr maybe we should check if fopen wrappers have been enabled and if not, use curl?
 http://php.net/manual/en/filesystem.configuration.php#ini.allow-url-fopen

Also, gzopen etc. functions might not work correctly on instances that run on shared hosts. Maybe we should explore usage of the WP Filesystem API.

@wpsmort do you have an instance running on share hosts conforming to the below description from https://codex.wordpress.org/Filesystem_API?

> Different hosting systems have different limitations in the way that their webservers are configured.
> 
> In particular, many hosting systems have the webserver running as a different user than the owner of the WordPress files. When this is the case, a process writing files from the webserver user will have the resulting files owned by the webserver's user account instead of the actual user's account. This can lead to a security problem in shared hosting situations, where multiple users are sharing the same webserver for different sites.
> 
> The WP_Filesystem is capable of detecting when the users for written files will not match, and switches to a method using FTP or similar instead. Depending on the available PHP libraries, WP_Filesystem supports three different methods of using FTP (via extension, sockets, or over-SSH) and will automatically choose the correct method.
> 
> In such a case, the plugin or theme implementing this code needs to request FTP credentials from the user. Functions have been added to make this easy to do and to standardize the look and feel of the credentials entry form.

Reply:
@contactashish13 It appeared to be a PHP change that causes an issue with compressed streams, and explained it more in detail on the issue ticket. https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/1428#issuecomment-354277191

There is an attempt with WP Filesystem API, but doesn't return the object https://github.com/EkoJr/all-in-one-seo-pack/blob/848745994fac2afce28070c7080f38b5ab4a359a/modules/aioseop_sitemap.php#L1003